### PR TITLE
Publish Bitbucket CI/CD Article

### LINKED ISSUE #1322: Article on CI/CD with Bitbucket
### Contact Details

@shippy

### This issue is being created to promote an approved Discussion.

- [X] I am promoting this issue from an approved Discussion and will link it below.

### Link to the Discussion this Issue is promoting.

#1215 

### If you're promoting this from an 'I want to write about...' Discussion, you can copy over the answers to the following questions. Otherwise, please answer these questions below

Early entry into the system this wanted went straight to an outline, but these are all answered well in the doc.

### Link to the initial outline we will develop through this Issue.

Doc contains context, outline, and draft:
https://docs.google.com/document/d/1SRZmUvYOTocrL04zZcQ5R0XZU0MatXM7LcUnMaf8Y94/edit?usp=sharing

### PR DESCRIPTION
**Pull request by gwenwindflower**
## Description & motivation
This PR adds a new article to the Developer Blog from @shippy covering specifically how to enable Slim CI/CD on Bitbucket, alongside the general principles for applying this method on any currently (offically) unsupported/unintegrated development platform (everything outside Github and GitLab)

This closes #1322

### CODE REVIEW

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (shippy) [inline website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: Bad single quotes

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (shippy) [inline website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: Bad single quotes

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (shippy) [inline website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: Should this be a code block?

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (shippy) [inline website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: ...I really can't think of another way to end this, but dear lord in heaven there must be a serious sentence somewhere out there, right? Alternately, I think this needs a gif to illustrate the "Ding!"

**Review by shippy — CHANGES_REQUESTED** [review]
The quotes are the only thing that would cause a problem for someone copy-pasting the code.

**Comment by gwenwindflower** [comment]
also need to add an author page for you @shippy so we can add the frontmatter metadata -- if you want to either send me the stuff you want here or feel free to just pull this branch and add a commit! here's the existing ones for an example: https://github.com/dbt-labs/docs.getdbt.com/blob/current/website/blog/authors.yml

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (shippy) [reply website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: Maybe this:

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (gwenwindflower) [reply website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: ehhh yea, yea, for consistency, even though it's one line it's still an entire file and we should make it a block.

**Comment by shippy** [comment]
Right - something like this:

```yaml
simon_podhajsky:
  name: Simon Podhajsky
  job_title: Data Lead
  organization: iLife Technologies
  description: >
    Simon Podhajsky is a lapsed neuroscientist turned full-stack data engineer 
    at iLife Technologies, a startup that seeks to digitize the life insurance 
    agency.
  image_url: /img/blog/authors/simon-podhajsky.png
  links:
    - url: https://www.linkedin.com/in/simonpodhajsky/
      icon: fa-linkedin
    - url: https://twitter.com/sim_pod
      icon: fa-twitter
```

I'll find a profile picture that fits and send it along.

**Comment by KiraFuruichi** [comment]
> also need to add an author page for you @shippy so we can add the frontmatter metadata -- if you want to either send me the stuff you want here or feel free to just pull this branch and add a commit! here's the existing ones for an example: https://github.com/dbt-labs/docs.getdbt.com/blob/current/website/blog/authors.yml

+1 to this and your author image! I believe you can also need to add a section to the beginning of the actual blog md where you have the title/author.etc. (example below). If you add this section, you can also remove the H1 # and text since the title tag will generate it for you (@gwenwindflower correct me if I'm wrong here)!

The other styling thing is adding a truncate somewhere in the beginning of your blog md file where you feel it's appropriate! Feel free to look at the other blog posts to see where they've implemented them.

image

**Comment by shippy** [comment]
> also need to add an author page for you @shippy so we can add the frontmatter metadata -- if you want to either send me the stuff you want here or feel free to just pull this branch and add a commit! here's the existing ones for an example: https://github.com/dbt-labs/docs.getdbt.com/blob/current/website/blog/authors.yml

@gwenwindflower I've made all the required changes but can't push to this repository -- wanna add me as a collaborator, or should I fork and make a PR to the PR?

File: website/blog/2022-04-14-add-ci-cd-to-bitbucket.md
Comment (gwenwindflower) [reply website/blog/2022-04-14-add-ci-cd-to-bitbucket.md]: hahahah YES there's no turning back now this is The One ✅ .

**Comment by gwenwindflower** [comment]
unfortunately @shippy i have to ask you to fork and PR your changes to my branch 😓 we have a limited number of seats for permissions for write access 

we're looking at some CMS options that might make this less of a pain in the ass -- with code it feels more understandable, but forking stuff to manage content definitely feels....extra...thanks for bearing with 🙏🏻

**Comment by shippy** [comment]
How lucky that I already have a fork for #1060 :) Will make the PR momentarily.

**Comment by gwenwindflower** [comment]
@johnblust this is ready to go whenever y'all are ready to publish 🙏🏻 -- feel free to merge away

**Comment by gwenwindflower** [comment]
closes #1322

**Comment by johnblust** [comment]
> @johnblust this is ready to go whenever y'all are ready to publish 🙏🏻 -- feel free to merge away

Thank you @gwenwindflower , this looks great! Thank you so much for the work you put into this ❤️  I'll need one of the reviewers to approve the PR before I am allowed to merge, though. @shippy an approved review from you might do the trick!

**Review by shippy — APPROVED** [review]
LGTM

**Comment by shippy** [comment]
@johnblust I submitted an approving review, but I think Github (correctly) blocks the merge because an authorized review by @runleonarun or @annafil is required.

**Comment by gwenwindflower** [comment]
@johnblust you technically actually can give it an approved review yourself since you have write access -- it's not limited to the folks autorequested, just anybody internal who has write access (this is why @shippy can't give it the ✅) but if process wise you're avoiding approving PRs, which i get, we could get @KiraFuruichi to 👍🏻 this one and lauren's ?

**Review by KiraFuruichi — APPROVED** [review]
Sooo excited for this one!

**Comment by gwenwindflower** [comment]
Screen Shot 2022-05-02 at 16 25 41

@johnblust if you ever want to review code that you haven't been requested on you can go to the `Files changed` tab and you'll see a button to add a review. You can give your opinion to anybody, who cares if they want it? You're John Blust!

**Comment by johnblust** [comment]
> Screen Shot 2022-05-02 at 16 25 41
> 
> @johnblust if you ever want to review code that you haven't been requested on you can go to the `Files changed` tab and you'll see a button to add a review. You can give your opinion to anybody, who cares if they want it? You're John Blust!

Learning something new everyday lol, thanks @gwenwindflower :)

**Comment by shippy** [comment]
So it seems like there might be three things that need to happen:

1. Change [the article date](https://github.com/dbt-labs/docs.getdbt.com/pull/1357/files#diff-8980d050f419bf0e688e2af9b87412f246418e5758e428cabd613769839f98a8R11) to sometime in May,
2. Rebase off base branch / merge it into this one,
3. Actually hit the Merge button while the date in (1) is still `CURRENT_DATE` and before anything else gets merged into `current` branch that would require re-doing (2).

@johnblust Is that something that you can do? Should I make another PR-to-this-PR?

**Comment by johnblust** [comment]
@shippy Yep! I've got it handled. 

Your article will be published this Friday, May 6th, on the Developer Blog sometime between 9am-10am ET. I can alert you via dbt Slack if you like! Then, on Monday, May 9th, we will share your article on our official LinkedIn and Twitter account at 8:30am ET if you would like to share/amplify with your own social pages. Also happy to alert you when the social push is live :)

Very excited to get this published for you!

**Comment by shippy** [comment]
@johnblust Thank you for taking care of it! No need for the separate alert, you're already doing plenty :)

**Pull request merged**
